### Importación de datos



In [19]:
import pandas as pd

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

tienda.head()

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  
0  Tarjeta de crédito                   8   4.60971 -74.08175  
1  Tarjeta de crédito                   4   6.25184 -75.56359  
2  Tarjeta de crédito                   1  10.39972 -75.51444  
3               Nequi                   1   3.43722 -76.52250  
4               Nequi                   1   6.25184 -75.56359

#1. Análisis de facturación



In [20]:
# --- ANÁLISIS 1: FACTURACIÓN TOTAL POR SUCURSAL ---

# El primer paso es entender el volumen de negocio. Calculamos los ingresos totales de cada sucursal.
# Para ello, sumamos todos los valores de la columna 'Precio'.

ingresos_sucursal1 = ventas_sucursal1['Precio'].sum()
ingresos_sucursal2 = ventas_sucursal2['Precio'].sum()
ingresos_sucursal3 = ventas_sucursal3['Precio'].sum()
ingresos_sucursal4 = ventas_sucursal4['Precio'].sum()

# Presentamos los resultados de forma clara y formateada para facilitar la comparación.
print("--- Reporte de Ingresos Totales por Sucursal ---")
print(f"Sucursal 1: ${ingresos_sucursal1:,.2f}")
print(f"Sucursal 2: ${ingresos_sucursal2:,.2f}")
print(f"Sucursal 3: ${ingresos_sucursal3:,.2f}")
print(f"Sucursal 4: ${ingresos_sucursal4:,.2f}")

--- Reporte de Ingresos Totales por Sucursal ---
Sucursal 1: $1,150,880,400.00
Sucursal 2: $1,116,343,500.00
Sucursal 3: $1,098,019,600.00
Sucursal 4: $1,038,375,700.00


# 2. Ventas por categoría

In [21]:
# --- ANÁLISIS 2: CATEGORÍAS DE PRODUCTOS MÁS POPULARES ---

# Contamos las ventas por categoría para cada sucursal y las ordenamos de mayor a menor.
top_categorias_s1 = ventas_sucursal1['Categoría del Producto'].value_counts()
top_categorias_s2 = ventas_sucursal2['Categoría del Producto'].value_counts()
top_categorias_s3 = ventas_sucursal3['Categoría del Producto'].value_counts()
top_categorias_s4 = ventas_sucursal4['Categoría del Producto'].value_counts()

# Mostramos las 3 categorías principales (el Top 3) para cada sucursal.
print("--- Top 3 Categorías Más Vendidas por Sucursal ---")
print("\n--- Sucursal 1 ---")
display(top_categorias_s1.head(3))
print("\n--- Sucursal 2 ---")
display(top_categorias_s2.head(3))
print("\n--- Sucursal 3 ---")
display(top_categorias_s3.head(3))
print("\n--- Sucursal 4 ---")
display(top_categorias_s4.head(3))

--- Top 3 Categorías Más Vendidas por Sucursal ---

--- Sucursal 1 ---


Categoría del Producto
Muebles         465
Electrónicos    448
Juguetes        324
Name: count, dtype: int64


--- Sucursal 2 ---


Categoría del Producto
Muebles         442
Electrónicos    422
Juguetes        313
Name: count, dtype: int64


--- Sucursal 3 ---


Categoría del Producto
Muebles         499
Electrónicos    451
Juguetes        315
Name: count, dtype: int64


--- Sucursal 4 ---


Categoría del Producto
Muebles         480
Electrónicos    451
Juguetes        338
Name: count, dtype: int64

# 3. Calificación promedio de la tienda


In [22]:
# --- ANÁLISIS 3: SATISFACCIÓN DEL CLIENTE (CALIFICACIÓN PROMEDIO) ---

# Calculamos la calificación promedio para cada sucursal usando la función .mean().
calif_promedio_s1 = ventas_sucursal1['Calificación'].mean()
calif_promedio_s2 = ventas_sucursal2['Calificación'].mean()
calif_promedio_s3 = ventas_sucursal3['Calificación'].mean()
calif_promedio_s4 = ventas_sucursal4['Calificación'].mean()

# Presentamos los promedios formateados a dos decimales para una lectura clara.
print("--- Calificación Promedio por Sucursal (escala de 1 a 5) ---")
print(f"Sucursal 1: {calif_promedio_s1:.2f} estrellas")
print(f"Sucursal 2: {calif_promedio_s2:.2f} estrellas")
print(f"Sucursal 3: {calif_promedio_s3:.2f} estrellas")
print(f"Sucursal 4: {calif_promedio_s4:.2f} estrellas")

--- Calificación Promedio por Sucursal (escala de 1 a 5) ---
Sucursal 1: 3.98 estrellas
Sucursal 2: 4.04 estrellas
Sucursal 3: 4.05 estrellas
Sucursal 4: 4.00 estrellas


# 4. Productos más y menos vendidos

In [23]:
# --- ANÁLISIS 4: PRODUCTOS ESTRELLA Y REZAGADOS ---

# Analizamos el rendimiento a nivel de producto individual.
# Usamos .value_counts() para obtener la lista ordenada de ventas por producto.
ventas_por_producto_s1 = ventas_sucursal1['Producto'].value_counts()
ventas_por_producto_s2 = ventas_sucursal2['Producto'].value_counts()
ventas_por_producto_s3 = ventas_sucursal3['Producto'].value_counts()
ventas_por_producto_s4 = ventas_sucursal4['Producto'].value_counts()

# Usamos .index[0] para el nombre del más vendido y .iloc[0] para su conteo.
# Usamos .index[-1] para el nombre del menos vendido y .iloc[-1] para su conteo.
print("--- Reporte de Productos Más y Menos Vendidos ---")
print("\n--- Sucursal 1 ---")
print("Producto Estrella:", ventas_por_producto_s1.index[0], f"({ventas_por_producto_s1.iloc[0]} ventas)")
print("Producto Rezagado:", ventas_por_producto_s1.index[-1], f"({ventas_por_producto_s1.iloc[-1]} ventas)")

print("\n--- Sucursal 2 ---")
print("Producto Estrella:", ventas_por_producto_s2.index[0], f"({ventas_por_producto_s2.iloc[0]} ventas)")
print("Producto Rezagado:", ventas_por_producto_s2.index[-1], f"({ventas_por_producto_s2.iloc[-1]} ventas)")

print("\n--- Sucursal 3 ---")
print("Producto Estrella:", ventas_por_producto_s3.index[0], f"({ventas_por_producto_s3.iloc[0]} ventas)")
print("Producto Rezagado:", ventas_por_producto_s3.index[-1], f"({ventas_por_producto_s3.iloc[-1]} ventas)")

print("\n--- Sucursal 4 ---")
print("Producto Estrella:", ventas_por_producto_s4.index[0], f"({ventas_por_producto_s4.iloc[0]} ventas)")
print("Producto Rezagado:", ventas_por_producto_s4.index[-1], f"({ventas_por_producto_s4.iloc[-1]} ventas)")

--- Reporte de Productos Más y Menos Vendidos ---

--- Sucursal 1 ---
Producto Estrella: Microondas (60 ventas)
Producto Rezagado: Celular ABXY (33 ventas)

--- Sucursal 2 ---
Producto Estrella: Iniciando en programación (65 ventas)
Producto Rezagado: Juego de mesa (32 ventas)

--- Sucursal 3 ---
Producto Estrella: Kit de bancas (57 ventas)
Producto Rezagado: Bloques de construcción (35 ventas)

--- Sucursal 4 ---
Producto Estrella: Cama box (62 ventas)
Producto Rezagado: Guitarra eléctrica (33 ventas)


# 5. Envío promedio por tienda

In [24]:
# --- ANÁLISIS 5: EFICIENCIA LOGÍSTICA (COSTO DE ENVÍO PROMEDIO) ---

# Finalmente, evaluamos si los costos operativos podrían ser un factor diferencial.
# Calculamos el costo promedio de envío para cada sucursal con .mean().

envio_promedio_s1 = ventas_sucursal1['Costo de envío'].mean()
envio_promedio_s2 = ventas_sucursal2['Costo de envío'].mean()
envio_promedio_s3 = ventas_sucursal3['Costo de envío'].mean()
envio_promedio_s4 = ventas_sucursal4['Costo de envío'].mean()

# Presentamos los resultados formateados como moneda para una fácil comparación.
print("--- Costo de Envío Promedio por Sucursal ---")
print(f"Sucursal 1: ${envio_promedio_s1:,.2f}")
print(f"Sucursal 2: ${envio_promedio_s2:,.2f}")
print(f"Sucursal 3: ${envio_promedio_s3:,.2f}")
print(f"Sucursal 4: ${envio_promedio_s4:,.2f}")

--- Costo de Envío Promedio por Sucursal ---
Sucursal 1: $26,018.61
Sucursal 2: $25,216.24
Sucursal 3: $24,805.68
Sucursal 4: $23,459.46


# 📄 Informe Ejecutivo: Análisis de Rendimiento de Sucursales Alura Store

**Para:** Señor Juan, Propietario de Alura Store  
**De:** Gonzalo Malca Garcia, Analista de Datos  
**Fecha:** 19 de agosto de 2025  
**Asunto:** Análisis de rendimiento y recomendación estratégica de venta de sucursal.

---

### **🎯 Resumen Ejecutivo y Recomendación Final**

Tras un análisis exhaustivo de las cuatro sucursales de Alura Store, basado en métricas clave de rendimiento, se ha llegado a una conclusión clara.

**Recomendación:** Basado en la evidencia recopilada, se recomienda proceder con la **venta de la Sucursal 4**. Esta decisión se fundamenta en su notablemente inferior facturación, que no se justifica por factores de calidad de servicio, popularidad de productos o eficiencia logística, áreas en las que compite adecuadamente con las demás sucursales.

---

### **🔎 Análisis Detallado de Métricas Clave**

A continuación, se presenta un resumen de los hallazgos de cada análisis realizado, los cuales sustentan la recomendación final.

#### **💰 Métrica 1: Facturación Total**
La facturación es el indicador más directo del rendimiento de negocio.

| Sucursal   | Ingresos Totales       |
| :--------- | :--------------------- |
| Sucursal 1 | `$1,150,880,400.00`    |
| Sucursal 2 | `$1,116,343,500.00`    |
| Sucursal 3 | `$1,098,019,600.00`    |
| **Sucursal 4** | **`$1,038,375,700.00`** |

*   **Insight:** La **Sucursal 4** es la que menos ingresos genera, estableciéndose como la principal candidata a ser analizada.

#### **📊 Métrica 2: Categorías Más Populares**
Este análisis verifica si todas las tiendas están alineadas con las preferencias del mercado.

*   **Insight:** Se confirmó que las tres categorías más populares (**Muebles, Electrónicos y Juguetes**) son las mismas en las cuatro sucursales. Esto indica que el bajo rendimiento de la Sucursal 4 no se debe a una mala oferta de productos.

#### **⭐ Métrica 3: Calificación Promedio del Cliente**
La satisfacción del cliente es clave para la lealtad y las ventas recurrentes.

*   **Insight:** Todas las sucursales gozan de una excelente reputación, con calificaciones promedio muy similares y cercanas a las **4.0 estrellas**. Por lo tanto, el problema de la Sucursal 4 no es la calidad del servicio.

#### **📈 Métrica 4: Productos Más y Menos Vendidos**
Un análisis más granular nos muestra el rendimiento a nivel de producto individual.

*   **Insight:** Cada sucursal tiene su propio "producto estrella", y los volúmenes de venta de estos son comparables. La Sucursal 4 es capaz de tener un producto exitoso, lo que sugiere que el problema no radica en productos específicos, sino en el volumen general.

#### **🚚 Métrica 5: Costo de Envío Promedio**
Los costos logísticos pueden afectar la rentabilidad.

*   **Insight:** Sorprendentemente, la **Sucursal 4 presenta los costos de envío promedio más bajos** (`$23,459.46`), demostrando ser la más eficiente en este aspecto. Esto descarta que su bajo rendimiento se deba a ineficiencias operativas.

---

### **🏁 Conclusión Final: Una Decisión Basada en Evidencia**

El perfil de la **Sucursal 4** es claro y consistente a través de todo el análisis:

*   ✅ **Vende** los tipos de productos correctos.
*   ✅ **Mantiene** a sus clientes felices.
*   ✅ **Opera** con la mayor eficiencia logística.
*   ❌ **Pero, a pesar de todo, factura significativamente menos.**

La conclusión lógica es que el bajo rendimiento de la Sucursal 4 se debe probablemente a factores externos no medidos en estos datos (como la ubicación, el tráfico de la zona o la competencia local) que limitan su volumen de ventas. Dado que el objetivo es obtener capital vendiendo la tienda con el rendimiento comercial más débil, la **Sucursal 4 es, sin lugar a dudas, la elección correcta.**